File: widgetssdk/src/main/java/com/glia/widgets/survey/SurveyState.java
package com.glia.widgets.survey;

import androidx.annotation.Nullable;

import java.util.List;

/**
 * @hide
 */
public class SurveyState {
    @Nullable
    public final String title;

    @Nullable
    public final List<QuestionItem> questions;

    private SurveyState(String title, List<QuestionItem> questions) {
        this.title = title;
        this.questions = questions;
    }

    static class Builder {
        @Nullable
        public String title;

        @Nullable
        public List<QuestionItem> questions;

        public Builder setTitle(@Nullable String title) {
            this.title = title;
            return this;
        }

        public Builder setQuestions(@Nullable List<QuestionItem> questions) {
            this.questions = questions;
            return this;
        }

        public Builder copyFrom(@Nullable SurveyState state) {
            if (state == null) {
                return this;
            }
            this.title = state.title;
            this.questions = state.questions;
            return this;
        }

        public SurveyState createSurveyState() {
            return new SurveyState(title, questions);
        }
    }
}

File: widgetssdk/src/main/java/com/glia/widgets/survey/SurveyState.kt
package com.glia.widgets.survey

/**
 * @hide
 */
internal class SurveyState private constructor(
    val title: String?,
    val questions: List<QuestionItem>?
) {
    internal class Builder {
        private var title: String? = null
        private var questions: List<QuestionItem>? = null

        fun setTitle(title: String?): Builder {
            this.title = title
            return this
        }

        fun setQuestions(questions: List<QuestionItem>?): Builder {
            this.questions = questions
            return this
        }

        fun copyFrom(state: SurveyState?): Builder {
            if (state == null) {
                return this
            }
            this.title = state.title
            this.questions = state.questions
            return this
        }

        fun createSurveyState(): SurveyState {
            return SurveyState(title, questions)
        }
    }
}
